codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ConfigProviderChain.java

```diff
@@ -109,7 +109,7 @@ public Builder staticProvider(String value) {
          * @return this builder
          */
         public Builder dynamicProvider(String rubyDefaultBlock) {
-            this.providers.add(new DynamicConfigProvider(rubyDefaultBlock));
+            this.providers.add(new DynamicConfigProvider("proc { |cfg| " + rubyDefaultBlock + " }"));
```

Comment:
Is this a block at this point?  Or is it a ruby code fragment or something like that?